service/accounting/db/query.sql

```diff
@@ -34,10 +34,18 @@ RETURNING *;
 SELECT id, company_id, branch_id, account_code, account_name, account_group, bank_name, 
 bank_account_number, bank_code, opening_balance, is_deleted
 FROM accounting.company_chart_of_accounts
-WHERE company_id = $1 AND account_name LIKE $2
-AND account_group LIKE $3 
+WHERE company_id = $1 
+AND account_name LIKE $2
+AND CASE 
+WHEN $5 = 'DEBET' THEN 
+is_deleted = FALSE AND (account_group = 'KAS' OR account_group = 'BANK')
+WHEN $5 = 'KREDIT' THEN 
+is_deleted = FALSE AND (account_group = 'BEBAN USAHA' OR account_group = 'BEBAN LAIN-LAIN')
+ELSE
+account_group LIKE $3 
 AND CASE WHEN $4 = TRUE OR $4 = FALSE THEN is_deleted = $4 
-ELSE is_deleted = is_deleted END;
+ELSE is_deleted = is_deleted END
```

Comment:
yang filter ini bisa tolong diginiin aja... ngecek pake filter / engga nya taruh di golang

`AND CASE WHEN @is_deleted_filter::bool THEN is_deleted = $4 ELSE TRUE END`

Reply:
Ok

Reply:
Ini setelah dicoba lagi gk bisa filter yang is_deleted false nantinya

Reply:
`AND CASE WHEN @is_deleted_filter::bool THEN is_deleted = $4 ELSE TRUE END`

kan nanti harusnya passing 2 variable... 
- is_deleted_filter (kalau true, tambah filter is_deleted = $4, kalau false ya AND TRUE)
- nilai is_deleted nya

is_deleted nya kok bisa jadi false?

Reply:
pertanyaan nya kenapa kok perlu is_deleted = FALSE ya? tujuannya pake soft delete apa? dari sisi client bakal bisa liat apa yang didelete terus mau direstore gitu? atau gimana?

Reply:
`SELECT
  id,
  company_id,
  branch_id,
  account_code,
  account_name,
  account_group,
  bank_name,
  bank_account_number,
  bank_code,
  opening_balance,
  is_deleted
FROM
  accounting.company_chart_of_accounts
WHERE
  company_id = $1
  AND account_name LIKE $2
  AND CASE
    WHEN @is_filter_account_group::bool THEN account_group IN $3::TEXT []
    ELSE TRUE
  END
  AND CASE
    WHEN @is_deleted_filter::bool THEN is_deleted = $4
    ELSE TRUE
  END
END;`

Reply:
`SELECT
  id,
  company_id,
  branch_id,
  account_code,
  account_name,
  account_group,
  bank_name,
  bank_account_number,
  bank_code,
  opening_balance,
  is_deleted
FROM
  accounting.company_chart_of_accounts
WHERE
  company_id = $1
  AND account_name LIKE $2
  AND CASE
    WHEN @is_filter_account_group::bool THEN account_group IN $3::TEXT[]
    ELSE TRUE
  END
  AND CASE
    WHEN @is_deleted_filter::bool THEN is_deleted = $4
    ELSE TRUE
  END
END;`

kalau yang ini engga sesuai nya kenapa ya? itu filter nya udah dipisah" jadi kalau mau tambah filter account group tinggal dikasih value filter account group waktu manggil query nya, kalau mau ditambah filter is deleted tinggal dikasih value filter is deleted nya...

Reply:
Ada 3 kemungkinan soalnya... lihat yg aktif, non aktif, sama all

Reply:
iya di query ku itu juga bisa aktif, non aktif, and all... 

is_action_group_filter,
action_group::TEXT[],
is_deleted_filter,
is_deleted,

btw non aktif, sama soft delete artinya beda lho... kalau mau active atau ga ya kasih is_active... brati harusnya selalu keluar tuh di response nya kalau is_deleted engga keluar...